modules/stochastic_tools/src/reporters/AdaptiveMonteCarloDecision.C

```diff
@@ -51,16 +54,15 @@ AdaptiveMonteCarloDecision::AdaptiveMonteCarloDecision(const InputParameters & p
   const auto rows = _sampler.getNumberOfRows();
   const auto cols = _sampler.getNumberOfCols();
 
-  // Create communicator that only has processors with rows
-  _communicator.split(
-      _sampler.getNumberOfLocalRows() > 0 ? 1 : MPI_UNDEFINED, processor_id(), _local_comm);
-
   // Initialize the required variables depending upon the type of adaptive Monte Carlo algorithm
   _inputs.resize(cols, std::vector<Real>(rows));
   _prev_val.resize(cols, std::vector<Real>(rows));
   _output_required.resize(rows);
   _prev_val_out.resize(rows);
 
+  // Check if GP is used as part of active learning
+  _gp_used = isParamValid("gp_decision") ? true : false;
+
```

Comment:
You can put this in the constructor initialization as:
```cpp
_gp_used(isParamValid("gp_decision")
```
You can then make this member variable `const`.